DesignSafe-CI/portal
# Task/DES-2022: Implement cache busting for webpack bundles.

## Overview: ##
Uses html-webpack-plugin to prevent stale webpack bundles from being imported. Each index HTML file that previously imported a webpack bundle has been replaced with a template file that Webpack converts to cache-busted html at compile time.

## PR Status: ##


## Related Jira tickets: ##

* [DES-2022](https://jira.tacc.utexas.edu/browse/DES-2022)

## Summary of Changes: ##
- Delete old HTML index files and add them to .gitignore
- Add template files for use with html-webpack-plugin
- Update webpack configuration to use html-webpack-plugin to generate index files with the correct hashes in the javascript/css imports.

## Testing Steps: ##
1. `npm install && npm run dev`
2. Click through designsafe.dev and confirm that there are no javascript/css import errors and the hashed filenames are being imported correctly: 
![image](https://user-images.githubusercontent.com/12601812/124833432-c83a5100-df43-11eb-967c-6d70f8ecc0c9.png)
3. `npm run build`
4. Click through again and make sure that the build succeeded and the new hashes are being imported correctly.

## UI Photos:

## Notes: ##
